### PR TITLE
Adjusting the authentication to the Bluemix platform.

### PR DESCRIPTION
**Pull request by isccarrasco**
Changing the mechanism for authenticating on the Bluemix platform, replacing the username and password mechanism to the apiKey mechanism as has been described on the official documentation.

### CODE REVIEW

File: iotp-adaptor/src/main/webapp/WEB-INF/web.xml [hunk 1]
```diff
@@ -12,17 +12,17 @@
...
     <description>Scheme used for URI when registering ServiceProvider.  Can be overridden by System property of the same name.</description>
     <param-name>com.ibm.oslc.adaptor.iotp.servlet.scheme</param-name>
-    <param-value>https</param-value>
+    <param-value>http</param-value>
   </context-param>
   <context-param>
     <description>Base URI for the adaptor.</description>
     <param-name>com.ibm.oslc.adaptor.iotp.servlet.baseurl</param-name>
-    <param-value>https://rlia4iot.raleigh.ibm.com:9443</param-value>
+    <param-value>http://192.168.1.91:8080</param-value>
```
Comment (berezovskyi) [inline iotp-adaptor/src/main/webapp/WEB-INF/web.xml]: I think localhost or 127.0.0.1 would be better.

**Comment by berezovskyi** [comment]
Thanks for the patch, @isccarrasco! I think it's better if @jamsden reviews this but it seems to be more than just an auth patch.

java.servlet-api is set to 4.0. I hope you know what you are doing. This is a API for Java EE 8 and requires newest servlet containers for such support https://stackoverflow.com/questions/46684980/jetty-and-java-ee-8 (strictly speaking).

All your import cleanups will be wiped out when the code generator is run again, I hope you also know that. Only the code inside the comment blocks starting with `Start of user code` is kept when you regenerate.

**Comment by berezovskyi** [comment]
@isccarrasco if you are using IntelliJ, it's best to use the following settings:

image

**Comment by berezovskyi** [comment]
Direct servlet support links:

https://www.eclipse.org/jetty/documentation/9.4.x/what-jetty-version.html
http://tomcat.apache.org/whichversion.html

File: iotp-adaptor/src/main/webapp/WEB-INF/web.xml (hunk 1 above)
Comment (isccarrasco) [reply iotp-adaptor/src/main/webapp/WEB-INF/web.xml]: Yes, sure, I forgot to reset those values. Thanks!

**Comment by isccarrasco** [comment]
> Thanks for the patch, @isccarrasco! I think it's better if @jamsden reviews this but it seems to be more than just an auth patch.
> 
> java.servlet-api is set to 4.0. I hope you know what you are doing. This is a API for Java EE 8 and requires newest servlet containers for such support https://stackoverflow.com/questions/46684980/jetty-and-java-ee-8 (strictly speaking).
> 
> All your import cleanups will be wiped out when the code generator is run again, I hope you also know that. Only the code inside the comment blocks starting with `Start of user code` is kept when you regenerate.

Ok, let me reset those values for only affect the auth process. Thanks @berezovskyi !

**Comment by jamsden** [comment]
I will try to look at these tomorrow

File: iotp-adaptor/src/main/webapp/rootservices_rdfxml.jsp [hunk 2]
```diff
@@ -40,8 +41,10 @@ String about = (String) request.getAttribute("about");
 	<oslc_am:amServiceProviders rdf:resource="<%= catalogUri %>" />
 	<oslc_rm:rmServiceProviders rdf:resource="<%= catalogUri %>" />
 	<oslc_cm:cmServiceProviders rdf:resource="<%= catalogUri %>" />
+	<oslc_cm:cmServiceProviders rdf:resource="<%= catalogUri %>" />
```
Comment (jamsden) [inline iotp-adaptor/src/main/webapp/rootservices_rdfxml.jsp]: Why do you have two cmServiceProviders with the same URI?

**Review by jamsden — CHANGES_REQUESTED** [review]
Can you add a link in your pull request description to the official documentation that's motivating the change to use api keys?

What are the container implications (for Jersey) when updating java.servlet-api to 4.0.1?
Why was junit changed from 4.13 to 4.12?
Remember to never change anything in generated code unless its inside generated // Start of user code and // End of user code. Note also that you cannot add these comments, they must be added in the templates used by the generator.
Any unused imports generated by the code generator should be submitted as change requests for the templates.
Can you use localhost instead of a fixed IP address?
Could you revert the organization of the imports in the generated code as these are unnecessary and clutter the pull request.

File: iotp-adaptor/src/main/webapp/rootservices_rdfxml.jsp (hunk 2 above)
Comment (isccarrasco) [reply iotp-adaptor/src/main/webapp/rootservices_rdfxml.jsp]: It was a mistake, I think I did a double copy when adding the `oslc_config:cmServiceProviders` in the rootservices jsp.

**Comment by jamsden** [comment]
[API ID Authentication](https://www.ibm.com/support/knowledgecenter/SSQP8H/iot/platform/reference/security/app_id.html) App ID is for authenticating users who need access to applications hosted in IBM Cloud. These users typically do not perform administrative or development activies on a cloud service and cannot log in to the platform service dashboard.  App ID is only for specific organizations.

iotp-adaptor needs to use [Cloud IAM Authentication and Authorization for Watson IoT Platform](https://www.ibm.com/support/knowledgecenter/SSQP8H/iot/platform/reference/security/cloud_iam.html?view=kc#cloud_iam) because users are doing development work and need to be able to log into the platform to get all the organizations (service providers) the user has access to.

At the moment, I do not see how a federated ID can use Cloud IAM for authentication. Still looking.

**Review by isccarrasco — COMMENTED** [review]
I have reverted all the changes within the classes that weren't modified for the implementation of the new mechanism for the authentication.

**Comment by berezovskyi** [comment]
@jamsden shall we merge?

**Review by jamsden — APPROVED** [review]

**Pull request merged**